lib/MainTemplate.js

```diff
@@ -354,6 +354,12 @@ module.exports = class MainTemplate extends Tapable {
 			buf.push("");
 			buf.push("// __webpack_public_path__");
 			buf.push(`${this.requireFn}.p = ${JSON.stringify(publicPath)};`);
+
+			if (outputOptions.libraryTarget === "system") {
```

Comment:
This code snippet should go into `SystemMainTemplatePlugin`.

You can tap into `mainTemplate.hooks.requireExtensions`